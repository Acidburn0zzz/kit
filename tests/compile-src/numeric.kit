include "stdio.h";

function main() {
    var a = 127;
    var b = 255;
    var c = 32767;
    var d = 100000;
    var e = 127_u64;
    var f = 127_f32;

    if ++a == -128 {
        printf("a overflow\n");
    }
    if ++b == -256 {
        printf("b overflow\n");
    }
    if ++c == -32768 {
        printf("c overflow\n");
    }
    if ++d == 100001 {
        printf("no d overflow\n");
    }
    if ++e == -128 {
        printf("e overflow\n");
    }
}
