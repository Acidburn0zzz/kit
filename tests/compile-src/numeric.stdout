a overflow
b overflow
c overflow
no d overflow
